teps/0080-taskrun-prepost-steps.md

```diff
@@ -0,0 +1,163 @@
+---
+status: proposed
+title: 'TaskRun: Pre/Post Steps'
+creation-date: '2021-08-18'
+last-updated: '2021-08-18'
+authors:
+- '@mattmoor'
+---
+
+# TEP-0080: TaskRun: Pre/Post Steps
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+This TEP allows users to partition their list of steps into three distinct phases
+(naming is "to-be-bikeshed"):
+ - `pre-steps`: a section of the TaskRun execution prior to the "real" work.  This
+ would generally consist of "setup" and "download" related work.
```

Comment:
This point is basically `init containers` right ? If we were to allow a user to inject init containers (pre-steps, whatever we call them), that would be feeling that need, right ?

Reply:
Yes, although Tekton has the capacity to do this pre-work with sidecars, which could be nice.